### PR TITLE
test/rbd/iscsi: correct the hostname in gwcli_create.t to match hostname -f

### PR DESCRIPTION
**Pull request by ideepika**
hostname -f and hostname generated from gwcli_create being different
gave rise to error:

The first gateway defined must be the local machine

This fixes regex to match `hostname -f` to avoid this failure.

Fixes: https://tracker.ceph.com/issues/53830

Signed-off-by: Deepika Upadhyay <dupadhya@redhat.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by ideepika** [comment]
https://pulpito.ceph.com/ideepika-2022-01-12_10:06:36-rbd:iscsi-wip-dis-testing-distro-default-smithi/

File: src/test/cli-integration/rbd/gwcli_create.t [hunk 1]
```diff
@@ -34,7 +34,7 @@ Create the target IQN
 
 Create the first gateway
 ========================
-  $ HOST=$(python3 -c "import socket; print(socket.getfqdn())" | cut -d. -f1)
+  $ HOST=$(python3 -c "import socket; print(socket.getfqdn())" | cut -f1)
```
Comment (idryomov) [inline src/test/cli-integration/rbd/gwcli_create.t:37]: IIRC the default delimiter for `cut` is a tab.  Does `socket.getfqdn()` return something with tabs in it?  If not, is `-f1` needed at all?

**Comment by idryomov** [comment]
I'm missing what changed between the two runs linked in the ticket that necessitates this PR:

good
http://qa-proxy.ceph.com/teuthology/teuthology-2022-01-03_02:01:03-rbd-master-distro-default-smithi/6591471/teuthology.log
bad
http://qa-proxy.ceph.com/teuthology/teuthology-2022-01-10_02:01:03-rbd-master-distro-default-smithi/6605019/teuthology.log

Could you please explain?

**Comment by ideepika** [comment]
sure, we are using dot as the delimiter here ` cut -d.` so the complete regex made fully qualified domain name(fqdn) to 
smithi001.front.sepia.ceph.com to smithi001; which I added IIRC, was hostname inside the docker, maybe something changed and we now have same hostname as the real hostname.

tldr; eg: generated(regex): smithi001  needed(hostname -f): smithi001.front.sepia.ceph.com

**Comment by idryomov** [comment]
> maybe something changed and we now have same hostname as the real hostname.

And this is exactly my question -- what was that change?

**Comment by ideepika** [comment]
> > maybe something changed and we now have same hostname as the real hostname.
> 
> And this is exactly my question -- what was that change?

not sure will check, @sebastian-philipp are you aware of any changes related to hostname for container?

File: src/test/cli-integration/rbd/gwcli_create.t (hunk 1 above)
Comment (ideepika) [reply src/test/cli-integration/rbd/gwcli_create.t:37]: socket.getfqdn() gives full domain name(hostname + domain name) I think we don't need cut at all now, will amend.

**Comment by ideepika** [comment]
without cut cmd passed: https://pulpito.ceph.com/ideepika-2022-01-12_12:17:43-rbd:iscsi-wip-dis-testing-distro-default-smithi

**Comment by idryomov** [comment]
... and without `cut` this is effectively a revert of "src/test/cli-integration: correct hostname for iscsi gateway" (https://github.com/ceph/ceph/commit/b09c6ca2cc639d702283d17e49366e191843dafc).  We need to understand what changed to avoid having to correct the correction of the correction (revert the revert?) again, a few months later.

**Comment by ideepika** [comment]
> ... and without `cut` this is effectively a revert of "src/test/cli-integration: correct hostname for iscsi gateway" ([b09c6ca](https://github.com/ceph/ceph/commit/b09c6ca2cc639d702283d17e49366e191843dafc)). We need to understand what changed to avoid having to correct the correction of the correction (revert the revert?) again, a few months later.

@djgalloway was something changed regarding the container host, probably around last week of Dec, don't see any direct affecting PR in ceph-ansible myself.

**Comment by idryomov** [comment]
Seems unlikely since it was merged on Dec 2 whereas the job was passing through all of December and also passed on Jan 3...

**Comment by idryomov** [comment]
jenkins test api

**Comment by idryomov** [comment]
jenkins test api

**Comment by ideepika** [comment]
What changed is `hostname -f` It has to be hostname change for the container either from lab setup or the way containers are deployed

**Comment by ideepika** [comment]
jenkins test api

**Comment by idryomov** [comment]
jenkins test make check

**Comment by idryomov** [comment]
jenkins test api

**Comment by idryomov** [comment]
jenkins test make check

**Comment by idryomov** [comment]
jenkins test make check

**Comment by ideepika** [comment]
jenkins test make check arm64

**Review by idryomov — APPROVED** [review]

**Comment by idryomov** [comment]
https://pulpito.ceph.com/dis-2022-01-14_08:49:53-rbd-wip-dis-testing-distro-default-smithi/
